File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.*;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        int[] answer = new int[commands.length];
        
        for (int i = 0; i < commands.length; ++i) {
            int st = commands[i][0];
            int en = commands[i][1];
            int k = commands[i][2];
            
            int[] arr = Arrays.copyOfRange(array, st - 1, en);
            Arrays.sort(arr);
            answer[i] = arr[k - 1];
        }
        return answer;
    }
}

File: 프로그래머스/1/42748. K번째수/K번째수.cpp
#include <bits/stdc++.h>
using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> answer;
    
    for (const auto& cmd : commands) {
        int st = cmd[0] - 1;
        int en = cmd[1] - 1;
        int k = cmd[2] - 1;
        vector<int> arr(array.begin() + st, array.begin() + en + 1);
        sort(arr.begin(), arr.end());
        
        answer.push_back(arr[k]);
    }
    return answer;
}